providers/implementations/rands/crngt.c

```diff
@@ -137,9 +156,9 @@ size_t prov_crngt_get_entropy(PROV_DRBG *drbg,
                                OSSL_SELF_TEST_DESC_RNG);
     }
 
-    while ((q = rand_pool_bytes_needed(pool, 1)) > 0 && attempts-- > 0) {
-        s = q > sizeof(buf) ? sizeof(buf) : q;
-        if (!crngt_get_entropy(libctx, crngt_glob->crngt_pool, buf, md, &sz))
+    for (t = bytes_needed; t > 0;) {
+        s = t > CRNGT_BUFSIZ ? CRNGT_BUFSIZ : t;
+        if (!crngt_get_entropy(drbg->provctx, crngt_glob->md, entp, md, &sz))
```

Comment:
Doesn't `crngt_get_entropy` always populate the supplied buffer with `CRNGT_BUFSIZ` bytes? If so isn't this going to overrun the buffer on the last iteration if `t < CRNGT_BUFSIZ`?

Reply:
Great catch.  Yes there is a risk here.  I don't think it's currently possible since everything is a multiple of CRNGT_BUFSIZ but better to make the code safe.